### PR TITLE
Extract profile feature module

### LINKED ISSUE #380: [Refactor] Separate existing profile code into :feature:profile
## Current state

Profile-related code is split across `:core:domain` and `:app`.

Current profile UI stubs and display code live in `app/src/main/java/timur/gilfanov/messenger/ui/screen/settings/`, including:

- `ProfileViewModel`
- `ProfileUiState`
- `ProfileUi`
- `ProfileSideEffects`
- `ProfileEditScreen`
- `ProfileEditViewModel`
- `ProfileEditUiState`
- `ProfileEditSideEffects`

Profile data contracts also already exist, but profile data implementation work for #336 would otherwise add more profile-specific data source, repository, DTO, Room, and DI code into `:app`.

## Desired state

Create `:feature:profile` and move existing profile-owned UI and profile data-layer implementation there before completing #336.

The resulting ownership should be:

- `:feature:profile` owns current-user global profile display/edit UI, ViewModels, profile data sources, profile repository implementation, profile DTOs, and profile-specific tests.
- `:core:domain` continues to own shared profile domain entities, repository/use-case contracts, and cross-module error types.
- `:app` remains the composition root: navigation host, app shell, `MessengerDatabase` assembly, and Hilt wiring that provides app-owned infrastructure to feature modules.

The feature module must not depend on `:app`.

## Motivation

This is needed before #336 grows the profile data layer inside `:app`.

ADR-001 and ADR-002 establish that feature modules should be created when feature development starts, not after all feature code has accumulated in the app module. Profile is now part of Phase 2 foundation work and is a prerequisite for implementing the profile-related requirements in `docs/Specification.md`, including profile display, profile editing, profile picture handling, and making profile changes visible in chats and message lists.

Separating profile in Phase 2 keeps ownership clear, reduces future extraction churn, and prevents `:app` from becoming the long-term home for profile UI/data/repository code.

This issue blocks #336 because #336 should implement the profile data layer in the final target module instead of adding more profile-specific implementation to `:app`.

## Affected layer

Multiple

## Estimated effort

M

## Suggested priority

high

## Behavior impact

- [x] This refactoring changes no externally observable behavior

## Affected files or modules

- Create `:feature:profile`
- Move profile UI/ViewModel/state/effect code from `app/src/main/java/timur/gilfanov/messenger/ui/screen/settings/`
- Move or create profile data-layer implementation in `:feature:profile`
- Update `settings.gradle.kts`, module Gradle config, Hilt bindings, and app navigation imports
- Keep `MessengerDatabase` composition in `:app`; expose only the profile DAO/infrastructure needed by `:feature:profile`
- Related issue: Blocks #336
- Related architecture docs: `docs/architecture/ADR-001-introduce-multi-module-architecture.md`, `docs/architecture/ADR-002-module-creation-strategy.md`

### PR DESCRIPTION
**Pull request by timurgilfanov**
Closes #380

## Summary
Extracts existing profile-owned UI, ViewModels, tests, use-case wiring, and profile data-source contracts into the new `:feature:profile` module so future profile data-layer work lands in its target feature module instead of `:app`.

## Changes
- Added `:feature:profile` with Compose, Hilt, Kover, ktlint/detekt, parcelize, resources, and test setup.
- Moved profile display/edit UI, state, side effects, ViewModels, tests, Hilt use-case wiring, and profile data-source contracts into the feature module.
- Updated app settings/navigation integration to consume profile screens from the feature module while keeping app as the composition root.
- Updated build, CI coverage/report wiring, and shared `repeatOnSubscription` ownership.

## Testing
- [x] Unit tests added or updated
- [ ] Manually tested
- [x] Edge cases considered

## Screenshots / Demo
Not applicable; no externally observable UI behavior intended to change.

## Acceptance criteria
- [x] Create `:feature:profile`.
- [x] Move existing profile UI/ViewModel/state/effect code from app settings.
- [x] Move profile data-source contracts into the profile feature.
- [x] Keep `:core:domain` as owner of shared profile domain APIs.
- [x] Keep `:app` as composition root and database assembly owner.
- [x] Ensure `:feature:profile` does not depend on `:app`.

## Breaking Changes
None.

## Follow-ups
- [ ] Implement profile data-layer behavior for #336 in `:feature:profile`.

### CODE REVIEW

File: feature/profile/src/main/kotlin/timur/gilfanov/messenger/profile/ui/screen/ProfileSection.kt
Comment (timurgilfanov) [inline feature/profile/src/main/kotlin/timur/gilfanov/messenger/profile/ui/screen/ProfileSection.kt:1]: Must add a preview

Comment (timurgilfanov) [reply feature/profile/src/main/kotlin/timur/gilfanov/messenger/profile/ui/screen/ProfileSection.kt:1]: Valid and in scope. Added `ProfileContent` previews for ready, loading, and German-locale states in `0f4c0f81`.

File: app/build.gradle.kts [hunk 1]
```diff
@@ -215,27 +217,29 @@ tasks.register<JacocoReport>("jacocoExternalCoverageReport") {
...
+            )
+        }
     }
     classDirectories.setFrom(
         fileTree(layout.buildDirectory.dir("tmp/kotlin-classes/$buildVariant")) {
             exclude(excludePatterns)
         } + fileTree(layout.buildDirectory.dir("intermediates/javac/$buildVariant/classes")) {
             exclude(excludePatterns)
-        } + featureAuthFileTrees,
+        } + featureFileTrees,
```
[the remark this answers is not in the document]
Comment (timurgilfanov) [reply app/build.gradle.kts]: Valid and in scope. Fixed in `0f4c0f81` by including the app build type in `featureVariants`, so feature-library `debug`/`release` class directories are considered instead of only full app variants.

File: app/src/main/java/timur/gilfanov/messenger/data/source/local/ResetSettingsLocalDataSourceError.kt [hunk 2]
```diff
@@ -1,3 +1,5 @@
 package timur.gilfanov.messenger.data.source.local
 
+import timur.gilfanov.messenger.profile.data.source.local.LocalDataSourceWriteError
+
 typealias ResetSettingsLocalDataSourceError = LocalDataSourceWriteError
```
[the remark this answers is not in the document]
Comment (timurgilfanov) [reply app/src/main/java/timur/gilfanov/messenger/data/source/local/ResetSettingsLocalDataSourceError.kt:5]: Valid concern, but out of scope for this PR. This PR intentionally follows the requested plan to keep `LocalDataSourceErrorV2`, `LocalDataSourceReadError`, and `LocalDataSourceWriteError` in `:feature:profile` rather than promoting them to `:core:data`; the final shared local data-source error model is tracked by #137.

File: feature/profile/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    debugImplementation(libs.androidx.ui.test.manifest)
+
+    implementation(project(":core:domain"))
+    implementation(project(":core:data"))
+    implementation(project(":core:ui"))
+    implementation(testFixtures(project(":core:domain")))
+    testImplementation(project(":core:test"))
+    testImplementation(testFixtures(project(":core:domain")))
+    testFixturesImplementation(project(":core:domain"))
+    testFixturesImplementation(testFixtures(project(":core:domain")))
```
[the remark this answers is not in the document]
Comment (timurgilfanov) [reply feature/profile/build.gradle.kts:110]: Valid and in scope. Removed the accidental `implementation(testFixtures(project(":core:domain")))` production dependency in `0f4c0f81`; the test fixture dependency remains scoped to tests/test fixtures.

File: feature/profile/src/main/kotlin/timur/gilfanov/messenger/profile/ui/screen/ProfileSection.kt [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            ProfileUiState.Loading -> "profile_loading"
+        }
+        Text(
+            text = stringResource(text),
+            style = MaterialTheme.typography.bodyMedium,
+            color = MaterialTheme.colorScheme.onBackground,
+            modifier = Modifier
+                .align(Alignment.Center)
+                .testTag(testTag),
+        )
```
[the remark this answers is not in the document]
Comment (timurgilfanov) [reply feature/profile/src/main/kotlin/timur/gilfanov/messenger/profile/ui/screen/ProfileSection.kt:86]: I verified this against `main`: the original `SettingsScreen.ProfileContent` rendered the same placeholder text for ready/loading and also left `onProfileEditClick` unused. So this specific restore request is not valid as a behavior-regression finding for #380. I did add the missing profile content previews in `0f4c0f81`.

File: feature/profile/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<resources>
+    <string name="settings_get_profile_failed">Cannot get profile</string>
+    <string name="settings_profile_ready_placeholder" translatable="false">Profile</string>
+    <string name="settings_profile_loading_placeholder" translatable="false">Profile loading...</string>
```
[the remark this answers is not in the document]
Comment (timurgilfanov) [reply feature/profile/src/main/res/values/strings.xml]: Valid and in scope. Fixed in `0f4c0f81` by making the strings translatable and adding German translations.

**Comment by timurgilfanov** [comment]
Outside-diff review outcome summary:

- `LocalDataSourceErrorV2` KDoc link: valid/in scope, fixed in `0f4c0f81` by converting the unresolved KDoc link to plain text.
- `LocalDataSourceErrorV2` throwable preservation: valid concern, out of scope for this extraction because the requested plan explicitly keeps `LocalDataSourceErrorV2` as-is in `:feature:profile`; final local data-source error modeling is tracked by #137.
- `RemoteProfileDataSource.updatePicture` upload contract: valid/in scope, fixed in `0f4c0f81` by documenting stream ownership, completion, failure, and cancellation behavior.
- Hardcoded `ObserveProfileUseCaseImpl` profile stub: valid concern, out of scope for this extraction; replacing the stub with the profile data layer is tracked by #336.
- `.github/workflows/main.yml` profile classfile artifacts: valid/in scope, fixed in `0f4c0f81` by uploading/downloading profile class files for coverage jobs.
- `ProfileEditUiState` / profile edit ViewModel state ownership: valid concern, out of scope because profile edit behavior was only moved by #380. I proposed a new follow-up issue and am waiting for issue-creation approval under the project rules.
- `ProfileViewModel` exhaustive `when`: not valid for the current sealed type; the current `when` handles the only `ObserveProfileError` case.
- Fully-qualified `Flow` import in the moved test: valid readability nit, but not a behavior/build issue; left unchanged in this pass.

File: settings.gradle.kts
Comment (timurgilfanov) [inline settings.gradle.kts:1]: Must add a file end symbol?

File: feature/profile/src/main/kotlin/timur/gilfanov/messenger/profile/data/source/remote/RemoteProfileDataSource.kt [hunk 6]
```diff
@@ -36,8 +36,11 @@ interface RemoteProfileDataSource {
     /**
      * Uploads and updates user's profile picture on the backend.
      *
-     * Emits upload progress updates through the returned Flow, allowing
-     * callers to track the upload status.
+     * The implementation consumes and closes [pictureStream] after use, including
+     * on failure or cancellation. The returned flow emits upload progress values;
+     * successful upload is represented by normal flow completion after the final
+     * progress update. Failures are emitted before completion, and cancelling the
+     * collector cancels the upload.
```
Comment (timurgilfanov) [inline feature/profile/src/main/kotlin/timur/gilfanov/messenger/profile/data/source/remote/RemoteProfileDataSource.kt]: Rollback KDoc update because this out of the scope of the PR.
```suggestion
     * Emits upload progress updates through the returned Flow, allowing
     * callers to track the upload status.
```

File: app/src/main/java/timur/gilfanov/messenger/ui/screen/settings/SettingsScreen.kt [hunk 7]
```diff
@@ -81,21 +63,24 @@ fun SettingsScreen(
...
-        onProfileEditClick = onProfileEditClick,
         onChangeLanguageClick = onChangeLanguageClick,
         onLogoutClick = { settingsViewModel.logout() },
         modifier = modifier,
     )
 }
 
 @Composable
-@Suppress("LongParameterList") // it's ok for screen to have large number of effects handlers
+@Suppress("LongParameterList")
```
Comment (timurgilfanov) [inline app/src/main/java/timur/gilfanov/messenger/ui/screen/settings/SettingsScreen.kt]: We don't need to suppress the check for 5 parameters.
```suggestion
```

File: app/src/main/java/timur/gilfanov/messenger/ui/screen/settings/SettingsScreen.kt [hunk 8]
```diff
@@ -81,21 +63,24 @@ fun SettingsScreen(
         }
     }
     SettingsScreenContent(
-        profileUiState = profileUiState,
+        profileContent = {
```
Comment (timurgilfanov) [inline app/src/main/java/timur/gilfanov/messenger/ui/screen/settings/SettingsScreen.kt:69]: Why make profile content parameter of `SettingsScreenContent`? We want to isolate settings UI for preview independently from profile UI? Do we need such isolation or we should preview completed settings UI as user will see it?

File: feature/profile/src/main/kotlin/timur/gilfanov/messenger/profile/ui/screen/ProfileSection.kt
Comment (timurgilfanov) [reply feature/profile/src/main/kotlin/timur/gilfanov/messenger/profile/ui/screen/ProfileSection.kt:1]: Must add preview with dark theme.

File: app/src/main/res/values/strings.xml
Comment (timurgilfanov) [inline app/src/main/res/values/strings.xml:1]: do we need end of the file symbol?

File: app/src/main/res/values-de/strings.xml
Comment (timurgilfanov) [inline app/src/main/res/values-de/strings.xml:1]: do we need end of the file symbol?

File: feature/profile/build.gradle.kts
Comment (timurgilfanov) [inline feature/profile/build.gradle.kts:1]: Must add comments similar to auth build config file.

File: feature/profile/build.gradle.kts [hunk 9]
```diff
@@ -0,0 +1,153 @@
+plugins {
+    alias(libs.plugins.android.library)
+    alias(libs.plugins.kotlin.android)
+    alias(libs.plugins.kotlin.compose)
+    alias(libs.plugins.detekt)
+    alias(libs.plugins.ktlint)
+    alias(libs.plugins.kover)
+    alias(libs.plugins.hilt)
+    alias(libs.plugins.ksp)
+    id("kotlin-parcelize")
```
Comment (timurgilfanov) [inline feature/profile/build.gradle.kts:10]: Do we need this?

File: .github/workflows/main.yml [hunk 10]
```diff
@@ -270,6 +282,7 @@ jobs:
             app/build/reports/tests/testMockDebugUnitTest
             core/domain/build/reports/tests/test
             feature/auth/reports/tests/testDebugUnitTest
+            feature/profile/build/reports/tests/testDebugUnitTest
```
Comment (timurgilfanov) [inline .github/workflows/main.yml:285]: Why this differ from line for `auth` feature: `feature/auth/reports/tests/testDebugUnitTest`? And pattern is the same with auth for `Run Component Tests` task.

File: settings.gradle.kts
Comment (timurgilfanov) [reply settings.gradle.kts:1]: I checked this locally: `settings.gradle.kts` already ends with a trailing newline (`0x0A`). No code change needed.

File: app/src/main/java/timur/gilfanov/messenger/ui/screen/settings/SettingsScreen.kt (hunk 8 above)
Comment (timurgilfanov) [reply app/src/main/java/timur/gilfanov/messenger/ui/screen/settings/SettingsScreen.kt:69]: I kept `profileContent` as a slot because `SettingsScreenContent` is still app-owned settings layout while profile state/effects belong to `:feature:profile`; this keeps app settings tests from owning profile ViewModel internals. The preview should still show what users see, so `24a9efbc` changed settings previews to pass the real feature `ProfileContent` instead of an empty placeholder.

File: app/src/main/res/values-de/strings.xml
Comment (timurgilfanov) [reply app/src/main/res/values-de/strings.xml:1]: I checked this locally: `app/src/main/res/values-de/strings.xml` already ends with a trailing newline (`0x0A`). No code change needed.
